src/nfts/seasons/ERC1155SeasonOne.sol

```diff
@@ -67,14 +94,17 @@ contract ERC1155SeasonOne is SeasonsBase {
         // Update total reward tokens needed in contract
         totalRewardTokens -= _rewardAmount;
 
+        // Update total reward tokens used
+        totalRewardTokensUsed += _rewardAmount;
+
         /// ---- interaction ---- ///
-        // Transfer
-        nftContract.safeTransferFrom(msg.sender, address(this), tokenId, 1, "");
 
         // Burn
-        nftContract.burn(address(this), tokenId, 1);
+        nftContract.burn(msg.sender, tokenId, 1);
```

Comment:
Have we thought about how this would work when the request is coming via the ThirdWeb (or other) paymaster? In that case, `msg.sender` wouldn't work.

Reply:
I think the solution is similar to what we do with `takePayment` on `GameConsumer.sol` (see [here](https://github.com/ZTX-Foundation/tuxedo/blob/a4516752f593e6d2c7d9f068e25d77a01fb3f98b/src/game/GameConsumer.sol#L93-L102)).